packages/usa-in-page-navigation/src/index.js

```diff
@@ -65,6 +65,39 @@ const getSectionAnchors = () => {
   return sectionAnchors;
 };
 
+/**
+ * Generates a unique ID for the given heading element.
+ *
+ * @param {HTMLHeadingElement} heading
+ *
+ * @return {string} - Unique ID
+ */
+const getHeadingId = (heading) => {
+  const baseId = heading.textContent
+    .toLowerCase()
+    // Replace non-alphanumeric characters with dashes
+    .replace(/[^a-z\d]/g, "-")
+    // Replace a sequence of two or more dashes with a single dash
+    .replace(/-{2,}/g, "-")
+    // Trim leading or trailing dash (there should only ever be one)
+    .replace(/^-|-$/g, "");
+
+  let id;
+  let suffix = 0;
+  do {
+    id = baseId;
+
+    // To avoid conflicts with existing IDs on the page, loop and append an
+    // incremented suffix until a unique ID is found.
+    suffix += 1;
+    if (suffix > 1) {
+      id += `-${suffix}`;
+    }
+  } while (document.getElementById(id));
```

Comment:
I really like this idea and could see it as its own utility in the future.

Only concern is it looks like we're looping over the headers twice? Since we're calling `getHeadingId(el)` inside of the `sectionHeadings.forEach((el)`.

Reply:
While this is technically a loop, I'd expect it would only run once in the vast majority of cases, since it would only ever run more than once if there was already an ID somewhere in the page which was the same as our computed ID using the heading's text.